opds.py

```diff
@@ -675,14 +675,22 @@ def add_entrypoint_links(cls, feed, url_generator, entrypoints,
         :param entrypoints: A list of all EntryPoints in the facet group.
         :param selected_entrypoint: The current EntryPoint, if selected.
         """
+        if (len(entrypoints) == 1
+            and selected_entrypoint in (None, entrypoints[0])):
+            # There is only one entry point. Unless the currently
+            # selected entry point is somehow different, there's no
+            # need to put any links at all here -- a facet group with
+            # one one facet might as well not be there.
+            return
+
         is_default = True
         for entrypoint in entrypoints:
             link = cls._entrypoint_link(
                 url_generator, entrypoint, selected_entrypoint, is_default,
                 group_name
             )
             if link:
-                cls.add_link_to_feed(feed, **link)
+                cls.add_link_to_feed(feed.feed, **link)
```

Comment:
This is a bug that wasn't caught by the test because the test created the wrong mock object.